src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+
+    #Initial Condition
+    initialcondition(Q, x...) = advection_sphere!(Q, DFloat(0), x...)
+    Q = MPIStateArray(spacedisc, initialcondition)
+
+    #Store Initial Condition as Exact Solution
+    Qe = copy(Q)
+
+    #Define Time-Integration Method
+    lsrk = LowStorageRungeKutta(spacedisc, Q; dt = dt, t0 = 0)
+
+    io = MPI.Comm_rank(mpicomm) == 0 ? stdout : devnull
+
+    #------------Set Callback Info--------------------------------#
+    # Set up the information callback
+    timer = [time_ns()]
+    cbinfo = GenericCallbacks.EveryXWallTimeSeconds(10, mpicomm) do (s=false)
+        if s
+            timer[1] = time_ns()
+        else
+            run_time = (time_ns() - timer[1]) * 1e-9
+            (min, sec) = fldmod(run_time, 60)
+            (hrs, min) = fldmod(min, 60)
+            @printf(io, "----\n")
+            @printf(io, "simtime =  %.16e\n", ODESolvers.gettime(lsrk))
+            @printf(io, "runtime =  %03d:%02d:%05.2f (hour:min:sec)\n", hrs, min, sec)
+            @printf(io, " Δmass =  %.16e\n", abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe) )
+        end
+        nothing
+    end
+
+    # Set up the information callback
+    cbmass = GenericCallbacks.EveryXSimulationSteps(1000) do
+        @printf(io, "----\n")
+        @printf(io, " Δmass =  %.16e\n", abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe) )
+        nothing
+    end
+
+    step = [0]
+    mkpath("vtk")
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1000) do (init=false)
+        outprefix = @sprintf("vtk/advection_sphere_%dD_mpirank%04d_step%04d",
+                             3, MPI.Comm_rank(mpicomm), step[1])
+        @printf(io, "----\n")
+        @printf(io, "doing VTK output =  %s\n", outprefix)
+        DGBalanceLawDiscretizations.writevtk(outprefix, Q, spacedisc, ("ρ", ))
+        step[1] += 1
+        nothing
+    end
+    #------------Set Callback Info--------------------------------#
+
+    #Perform Time-Integration
+    solve!(Q, lsrk; timeend=timeend, callbacks=(cbinfo, cbmass, cbvtk))
+
+    # Print some end of the simulation information
+    if integration_testing
+        error = euclidean_distance(Q, Qe) / norm(Qe)
+        Δmass = abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+        @info @sprintf """Finished
+        error = %.16e
+        Δmass = %.16e
+        """ error Δmass
+    end
+
+    #return diagnostics
+    return (error, Δmass)
+end
+#}}} Main
+
+#{{{ Run Script
+function run(mpicomm, Nhorizontal, Nvertical, N, timeend, DFloat, dt, ArrayType)
+    Rrange=range(DFloat(1); length=Nvertical+1, stop=2)
+    topl = StackedCubedSphereTopology(mpicomm,Nhorizontal,Rrange; boundary=(0,0))
+    (error, Δmass) = main(mpicomm, DFloat, topl, N, timeend, ArrayType, dt)
+end
+#}}} Run Script
+
+#{{{ Run Program
+using Test
+let
+    MPI.Initialized() || MPI.Init()
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    if MPI.Comm_rank(mpicomm) == 0
```

Comment:
I'd recommend change this whole if-else block to:

```
  ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
  loglevel = ll == "DEBUG" ? Logging.Debug :
  ll == "WARN"  ? Logging.Warn  :
  ll == "ERROR" ? Logging.Error : Logging.Info
  global_logger(ConsoleLogger(MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull,
                              loglevel)
```

It's a bit more MPI friendly as then all the logging calls will get called on all the MPI ranks (not just the root)